scene/3d/ray_cast_3d.cpp

```diff
@@ -148,14 +148,14 @@ void RayCast3D::_notification(int p_what) {
 		case NOTIFICATION_ENTER_TREE: {
 			if (enabled && !Engine::get_singleton()->is_editor_hint()) {
 				set_physics_process_internal(true);
-
-				if (get_tree()->is_debugging_collisions_hint()) {
-					_update_debug_shape();
-				}
 			} else {
 				set_physics_process_internal(false);
 			}
```

Comment:
This could become:
```
set_physics_process_internal(enabled && !Engine::get_singleton()->is_editor_hint());
```

Reply:
Perhaps some extra logic could be added in the future, I think it's not performance critical in either case.